scipio/src/executor.rs

```diff
@@ -751,13 +765,25 @@ impl LocalExecutor {
             // requests that are latency sensitive we want them out of the
             // ring ASAP (before we run the task queues). We will also use
             // the opportunity to install the timer.
-            let duration = self.preempt_timer_duration();
-            self.parker.poll_io(duration);
+            self.parker.poll_io(preempt);
+
             if !self.run_one_task_queue() {
                 self.parker.park();
             }
         })
     }
+
+    /// Polls f and measures its time.
+    fn measure_poll<T>(
+        stats: &mut ExecutorStats,
+        future: Pin<&mut impl Future<Output = T>>,
```

Comment:
I am not sure about it to whoever will be reviewing this

Reply:
This is not the right approach.

As I said above, the main future that we poll here will essentially only be ready when the whole program is ready. In a long-living server, that might mean never. We need to account task queues, no futures.

You could try to measure *every* poll, but that's too expensive. So again, the right thing to do is to measure task queue time. Once we resort to measuring task queue time, there may be time unaccounted because it was run in the main future. For that we can add something like you are adding here. But it is a complement, not the whole thing.

Reply:
Ok, I moved to the simplest implementation possible for spinning only. This piece no longer exists.